core-contracts/contracts/process_template.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -euo pipefail
+
+if (( $# < 3 )); then
+    echo "Usage: $0 <input-directory> <output-directory> <yaml-files...>"
+    exit 1
+fi
+
+in_dir="$1"
+shift 1
+
+out_dir="$1"
+shift 1
+
+# The rest of the args will be treated as YAML files
+yaml_files=("$@")
+shift $#
+
+# Create tmpfile for YAML, and remove it when script exits
+yaml_tmp=$(mktemp --suffix .yaml)
+trap 'rm -f -- "$yaml_tmp"' INT TERM HUP EXIT
+
+# Combine all YAML
+echo "Using YAML files: ${yaml_files[*]}"
+cat "${yaml_files[@]}" > "$yaml_tmp"
+
+# Get list of files, null separated
+file_list=$(find "$in_dir" -type f -printf '%P\n')
```

Comment:
For compatibility with BSD `find`, this can instead be:
```suggestion
file_list=$(find "$in_dir" -type f | sed "s|^$in_dir/||")
```